util/util.go

```diff
@@ -0,0 +1,42 @@
+package util
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha512"
+	"io"
+
+	"github.com/flynn/go-tuf/data"
+)
+
+type BytesReadCloser struct {
+	*bytes.Reader
+}
+
+func (b BytesReadCloser) Close() error {
+	return nil
+}
+
+func FileMetaEqual(actual data.FileMeta, expected data.FileMeta) bool {
+	if actual.Length != expected.Length {
+		return false
+	}
+	for typ, hash := range expected.Hashes {
+		if !hmac.Equal(actual.Hashes[typ], hash) {
```

Comment:
What's the behavior if for example comparing the `actual` file metadata produced by the currently executing process against the `expected` file metadata deserialized from a different version, and the `expected` has hash types the currently running version hasn't implemented?